### PR TITLE
samples: updating overlays of openthread samples

### PR DESCRIPTION
**Pull request by canisLupus1313**
Remove the common folder from openthread samples.

### CODE REVIEW

**Review by wiba-nordic — APPROVED** [review]
Looks good.

File: samples/openthread/common/overlay-thread_1_2_ftd.conf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
```
Comment (lmaciejonczyk) [inline samples/openthread/common/overlay-thread_1_2_ftd.conf:2]: it should be 2021 since it's new file

**Review by lmaciejonczyk — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_thread_certification.rst [hunk 2]
```diff
@@ -63,16 +63,19 @@ Complete the following steps to prepare for the certification tests:
...
-   The :ref:`ot_cli_sample` sample is used as a base, modified with the :file:`harness/overlay-cert.conf` overlay file.
+   The :ref:`ot_cli_sample` sample is used as a base, modified with the following overlay files:
+   * :file:`../common/overlay-no-logging.conf`
+   * :file:`../common/overlay-ble.conf`
 
    .. code-block::
 
          cd ncs/nrf/samples/openthread/cli/
-         west build -b nrf52840dk_nrf52840 -- -DOVERLAY_CONFIG=harness/overlay-cert.conf -DCONFIG_OPENTHREAD_LIBRARY_1_1=y
+         west build -b nrf52840dk_nrf52840 -- -DOVERLAY_CONFIG="../common/overlay-no-logging.conf;../common/overlay-ble.conf" -DCONFIG_OPENTHREAD_LIBRARY_1_1=y
```
Comment (tejlmand) [inline doc/nrf/ug_thread_certification.rst:73]: Curious, why is Bluetooth enabling required for Thread certification ?
That is not clear to me at all, and thus I doubt the users understands why bluetooth must be enabed.

File: doc/nrf/ug_thread_certification.rst [hunk 3]
```diff
@@ -63,16 +63,19 @@ Complete the following steps to prepare for the certification tests:
...
 
    .. note::
       The configuration option selects the precompiled OpenThread libraries.
-      The overlay file enables :ref:`multiprotocol support <ug_multiprotocol_support>` with Bluetooth® LE advertising.
+      The overlay :file:`../common/overlay-no-logging.conf` disables logging.
```
Comment (tejlmand) [inline doc/nrf/ug_thread_certification.rst:77]: Instead of a `no-logging.conf` overlay, should this be turned into a:
1. `prj.conf` (release ready build)
2. `prj_debug.conf` (debug build)

See more here: https://docs.zephyrproject.org/latest/guides/build/kconfig/setting.html#the-initial-configuration

File: doc/nrf/ug_thread_certification.rst [hunk 4]
```diff
@@ -63,16 +63,19 @@ Complete the following steps to prepare for the certification tests:
 
 #. Build the certification image.
 
-   The :ref:`ot_cli_sample` sample is used as a base, modified with the :file:`harness/overlay-cert.conf` overlay file.
+   The :ref:`ot_cli_sample` sample is used as a base, modified with the following overlay files:
+   * :file:`../common/overlay-no-logging.conf`
+   * :file:`../common/overlay-ble.conf`
```
Comment (tejlmand) [inline doc/nrf/ug_thread_certification.rst:68]: Not sure placing Kconfig fragments like this is the best approach.
It looks nice to use, but it makes it much harder for the users to know where to look for additional configurations for a given sample.

For example, I can look here:
https://github.com/nrfconnect/sdk-zephyr/tree/master/samples/net/sockets/echo_server
And I immediately know that there exists additional configuration options for the echo_server, like bluetooth, openthread, 802154, etc.

With this approach, it will be much harder for users to know where to look.
Also, are all Kconfig fragments suitable for all samples ?
Does it make sense to use `overlay-thread_1_2_ftd.conf` with the https://github.com/nrfconnect/sdk-nrf/tree/master/samples/openthread/coap_client sample ?

File: samples/openthread/common/overlay-ble.conf [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Enable Bluetooth LE controller which supports multiprotocol
+CONFIG_BT_LL_SOFTDEVICE_DEFAULT=y
+
+# Enable and configure Bluetooth LE
+CONFIG_BT=y
+CONFIG_BT_PERIPHERAL=y
+CONFIG_BT_DEVICE_NAME="NCS DEVICE"
```
Comment (tejlmand) [inline samples/openthread/common/overlay-ble.conf:13]: This means that if the user apply this Kconfig fragment to 2 thread devices for testing, then both shows up with identical names.

To me that indicates the samples that should support multiprotocol should have their own fragment so they are easier to identify, like `ncs coap client`.
Btw. Why only upper case in `NCS DEVICE` ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
Not convinced that a common folder for Kconfig fragments like this is the best approach.

Main reason is that it will be even harder for our users to understand where to look for configurations applicable for a given sample exists.
nRF Connect SDK is already very complex for our users, so let's try to avoid making it even harder.
Openthread proposes `sdk-nrf/samples/openthread/common`.
Applications, matter, bluetooth samples could all start making their own common folders, all leading to harder to understand for our users.

**Comment by tejlmand** [comment]
> 
> Signed-off-by: Przemyslaw Bida [przemyslaw.bida@nordicsemi.no](mailto:przemyslaw.bida@nordicsemi.no)

This is the PR message ?
What is this PR about.

>This commit moves overlays of openthread cli sample to more generic
loation.

If looking into history to spot a potential reason for a problem or having to do a bisect, then a bit more descriptive commit message would be nice.

Also:
1. Why is this being done.
2. What has changed.

File: doc/nrf/ug_thread_certification.rst (hunk 4 above)
Comment (Rallare) [reply doc/nrf/ug_thread_certification.rst:68]: As a generic note, we should try to align with other samples and applications in our sdk (and its deps), including the tree structure. I support the comment and thoughts from @tejlmand here.

Comment (OeySa) [reply doc/nrf/ug_thread_certification.rst]: I concur with @Rallare

Comment (canisLupus1313) [reply doc/nrf/ug_thread_certification.rst]: Take a look at the structure now https://github.com/canisLupus1313/sdk-nrf/tree/thread_1_2_overlay_move/samples/openthread/cli

File: doc/nrf/ug_thread_certification.rst (hunk 2 above)
Comment (canisLupus1313) [reply doc/nrf/ug_thread_certification.rst]: Those two things are Using the same radio peripheral and we want to be sure that cooperation between them is works properly.

File: samples/openthread/common/overlay-ble.conf (hunk 5 above)
Comment (canisLupus1313) [reply samples/openthread/common/overlay-ble.conf]: > Btw. Why only upper case in NCS DEVICE ?
This is taken from original file why it was like that there I don't know.

File: samples/openthread/cli/prj.conf [hunk 6]
```diff
@@ -15,3 +15,14 @@ CONFIG_SHELL_CMD_BUFF_SIZE=416
 
 # Enable OpenThread features set
 CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER=y
+
+CONFIG_NET_L2_OPENTHREAD=y
+
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_CONNECTION_MANAGER=y
```
Comment (edmont) [inline samples/openthread/cli/prj.conf:23]: `CONFIG_NET_CONNECTION_MANAGER` can be removed from all of our samples since we are not using it.

File: samples/openthread/coprocessor/README.rst [hunk 7]
```diff
@@ -116,6 +116,7 @@ For more information about using configuration overlay files, see :ref:`zephyr:i
 
 The following configuration files are available:
 
+* :file:`overlay-thread_1_2_ftd.conf` Enables experimantal Thread 1.2 extension.
```
Comment (edmont) [inline samples/openthread/coprocessor/README.rst:119]:
```suggestion
* :file:`overlay-thread_1_2_ftd.conf` Enables experimental Thread 1.2 extension.
```

File: samples/openthread/coprocessor/overlay-thread_1_2_ftd.conf [hunk 8]
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Configure dependencies
+CONFIG_NRF_802154_ENCRYPTION=y
+CONFIG_IEEE802154_2015=y
+CONFIG_IEEE802154_CSL_ENDPOINT=y
```
Comment (edmont) [inline samples/openthread/coprocessor/overlay-thread_1_2_ftd.conf:10]: `CONFIG_IEEE802154_CSL_ENDPOINT` not required for FTD.

File: samples/openthread/cli/overlay-thread_1_2_ftd.conf [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Configure dependencies
+CONFIG_NRF_802154_ENCRYPTION=y
+CONFIG_IEEE802154_2015=y
+CONFIG_IEEE802154_CSL_ENDPOINT=y
```
Comment (edmont) [inline samples/openthread/cli/overlay-thread_1_2_ftd.conf:10]: `CONFIG_IEEE802154_CSL_ENDPOINT` not required for FTD.

**Review by edmont — APPROVED** [review]
Thanks!

Please stay within 72 chars for commit messages.

File: doc/nrf/ug_thread_certification.rst (hunk 3 above)
Comment (canisLupus1313) [reply doc/nrf/ug_thread_certification.rst]: So Idea behind no logging overlay is not for release or debug purposes but rather turning off prompt, banner, and excessive logging that are interfering with Thread certification. I don't say that having two project configs Debug and Release is bad idea but this overlay wasn't intended for Release purposes.

File: subsys/net/openthread/Kconfig.defconfig [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+	int
+	default 16
+
+config NET_BUF_TX_COUNT
+	int
+	default 16
+
+config OPENTHREAD_DEBUG
+	bool
+	default y
```
Comment (LuDuda) [inline subsys/net/openthread/Kconfig.defconfig:91]: Why debug feature is enabled by default?

File: subsys/net/openthread/Kconfig.defconfig [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+config OPENTHREAD_L2_DEBUG
+	bool
+	default y
```
Comment (LuDuda) [inline subsys/net/openthread/Kconfig.defconfig:95]: Why debug feature is enabled by default?

**Comment by greg-fer** [comment]
@canisLupus1313 , @edmont , I hear rumours this PR is for 1.7.0. If so, it is missing the milestone and might not make it... FYI, @shantha-14

File: samples/openthread/cli/overlay-thread_1_2_ftd.conf [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Configure dependencies
+CONFIG_NRF_802154_ENCRYPTION=y
+CONFIG_IEEE802154_2015=y
+CONFIG_NET_PKT_TXTIME=y
+CONFIG_NET_PKT_TIMESTAMP=y
+CONFIG_OPENTHREAD_MAC_SOFTWARE_TX_SECURITY_ENABLE=n
+CONFIG_OPENTHREAD_SLAAC=y
```
Comment (LuDuda) [inline samples/openthread/cli/overlay-thread_1_2_ftd.conf:13]: Don't we want enable SLAAC by default?

**Review by LuDuda — COMMENTED** [review]
Looking at CI, this PR affects Matter and may also others like HomeKit or Pelion so we need a careful review (@maciejpietras @pdunaj)

We're looking at .config differences with and without this change.

File: samples/openthread/cli/overlay-thread_1_2_ftd.conf (hunk 12 above)
Comment (edmont) [reply samples/openthread/cli/overlay-thread_1_2_ftd.conf:13]: It's added here as a dependency for DUA. It's also added by `CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` for `cli` and `coprocessor` samples, since it's required for certification. The CoAP samples don't include it since they are intended to be minimal feature use cases.

**Comment by tomaszkob89** [comment]
Together with this PR we need to merge https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-thread/pull-requests/382/overview (adaptations for test repository) to not break our CI

File: subsys/net/openthread/Kconfig.defconfig [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+
+config NET_CONNECTION_MANAGER
+	bool
+	default y
+
+config MAIN_STACK_SIZE
+	int
+	default 2048
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:35]: Why and when is a stack size of `2048` in the cases where it's not already `2048` per default ?

stack size is already 2048 when using cc3xx entropy:
https://github.com/nrfconnect/sdk-nrf/blob/612928cec9588b41357139771ae2046ad7183e8c/Kconfig.nrf#L36

Which is also default here:
https://github.com/nrfconnect/sdk-nrf/blob/612928cec9588b41357139771ae2046ad7183e8c/drivers/entropy/Kconfig#L7-L12

**Comment by greg-fer** [comment]
Remember to update the changelog (if really needed)

File: subsys/net/openthread/Kconfig.defconfig [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+	bool
+	default n
+
+config NET_IPV6_MLD
+	bool
+	default n
+
+config NET_TX_STACK_SIZE
+	int
+	default 1200
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:63]: Already default to 1200 in Zephyr, why is it needed here ?
https://github.com/nrfconnect/sdk-zephyr/blob/66ef6a7e125d53b5d650effcce101ab901627100/subsys/net/ip/Kconfig.stack#L10
```
config NET_TX_STACK_SIZE
	int "TX thread stack size"
	default 1200
```

File: subsys/net/openthread/Kconfig.defconfig [hunk 15]
```diff
@@ -0,0 +1,149 @@
...
+
+config NET_RX_STACK_SIZE
+	int
+	default 1500
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:67]: Same here:
https://github.com/nrfconnect/sdk-zephyr/blob/66ef6a7e125d53b5d650effcce101ab901627100/subsys/net/ip/Kconfig.stack#L18

```
config NET_RX_STACK_SIZE
	int "RX thread stack size"
	default 1500
```

File: subsys/net/openthread/Kconfig.defconfig [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+	int
+	default 10
+
+config NET_PKT_TX_COUNT
+	int
+	default 16
+
+config NET_BUF_RX_COUNT
+	int
+	default 16
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:79]: Same as upstream default:
https://github.com/nrfconnect/sdk-zephyr/blob/66ef6a7e125d53b5d650effcce101ab901627100/subsys/net/ip/Kconfig#L638

File: subsys/net/openthread/Kconfig.defconfig [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+
+config NET_BUF_TX_COUNT
+	int
+	default 16
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:83]: also same as upstream

File: subsys/net/openthread/Kconfig.defconfig [hunk 18]
```diff
@@ -0,0 +1,149 @@
...
+
+config OPENTHREAD_MANUAL_START
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:85]: seems to be same as upstream

File: subsys/net/openthread/Kconfig.defconfig [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+
+config MBEDTLS_CIPHER_MODE_CBC
+	bool
+	default n
+
+config MBEDTLS_CIPHER_MODE_CTR
+	bool
+	default n
+
+config MBEDTLS_CIPHER_MODE_CFB
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:101]: Could you checkup on this, doesn't seem to be needed.

File: subsys/net/openthread/Kconfig.defconfig [hunk 20]
```diff
@@ -0,0 +1,149 @@
...
+	bool
+	default n
+
+config MBEDTLS_CIPHER_MODE_OFB
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:105]: From here and below I have not verified the symbols, could you please do that yourself.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please checkup on the symbols:
- Remove those settings that are already having a proper default value.
- Make comments on all symbols where default is changed, as to why another default is better.

File: subsys/net/openthread/Kconfig.defconfig (hunk 19 above)
Comment (canisLupus1313) [reply subsys/net/openthread/Kconfig.defconfig:74]: This is not used by our samples so it's just to decrease size.

File: subsys/net/openthread/Kconfig.defconfig (hunk 20 above)
Comment (canisLupus1313) [reply subsys/net/openthread/Kconfig.defconfig:78]: # This is needed to properly handle MPSL in openthread samples

File: subsys/net/openthread/Kconfig.defconfig (hunk 18 above)
Comment (canisLupus1313) [reply subsys/net/openthread/Kconfig.defconfig:58]: This is true only in when OPENTHREAD_COPROCESSOR is turned on.

**Review by tejlmand — APPROVED** [review]
Approved, as we have a followup plan for the outstanding issues.

File: subsys/net/openthread/Kconfig.defconfig (hunk 20 above)
Comment (tejlmand) [reply subsys/net/openthread/Kconfig.defconfig:78]: That is not what I meant with that comment.

What I meant was that I had not been through all the symbols to see if those are already unavailable or default `n` in a default setup.

I found several symbols that already defaulted to the desired value, if you check the comments above, but asked you to carry out the same check for symbols below.

File: subsys/net/openthread/Kconfig.defconfig (hunk 19 above)
Comment (tejlmand) [reply subsys/net/openthread/Kconfig.defconfig:74]: but as you are defaulting to use CryptoCell and afaik only officially support OpenThread on hw with CC, then this symbol is not available, because it relies on software only mbedTLS or Oberon.

File: subsys/net/openthread/Kconfig.defconfig [hunk 21]
```diff
@@ -0,0 +1,114 @@
...
+# This file only changes defaults and thus all symbols here must be prompltless
+# and safeguarded so that they only are applied when building Thread.
+
+if NET_L2_OPENTHREAD
+
+choice LIBC_IMPLEMENTATION
+	default NEWLIB_LIBC
+endchoice
+
+# This is needed to properly handle MPSL in openthread samples
```
Comment (tejlmand) [inline subsys/net/openthread/Kconfig.defconfig:17]: and exactly why is this needed in OpenThread samples ?
https://github.com/nrfconnect/sdk-zephyr/blob/master/kernel/Kconfig#L100-L118
> 	int "Number of very-high priority 'preemptor' threads"

and
 
>This defines a set of priorities at the (numerically) lowest
> end of the range which have "meta-irq" behavior.  Runnable
> threads at these priorities will always be scheduled before
> threads at lower priorities, EVEN IF those threads are
> otherwise cooperative and/or have taken a scheduler lock.

What in openthread has this need ? and why is it needed ?
I think we can be a bit more informative to our users here.

**Comment by SebastianBoe** [comment]
@canisLupus1313 : Can you investigate why CI is red ?

**Comment by SebastianBoe** [comment]
> Together with this PR we need to merge https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-thread/pull-requests/382/overview (adaptations for test repository) to not break our CI

I don't know what that means @tomaszkob89 . Do we await merging this until something else is merged?

**Comment by canisLupus1313** [comment]
@SebastianBoe  this is moving one of the files that are used in thread CI however the config of this CI resided in different repo.

**Comment by shantha-14** [comment]
Removed milestone as testing is not complete. Agreed with @MarekPorwisz

**Comment by greg-fer** [comment]
Adding @Avalei as this PR is related to VSCode support.

**Review by Avalei — APPROVED** [review]
> 
> 
> Adding @Avalei as this PR is related to VSCode support.

@greg-fer I do not see any sections related to VSC, perhaps the file was removed and no longer relevant. We have a separate task to update the NCS sample template, which may affect this later, but for now I will approve this PR.

**Review by rlubos — APPROVED** [review]
